opensearch-aws-sigv4.gemspec

```diff
@@ -50,5 +49,5 @@ Gem::Specification.new do |s|
   s.required_ruby_version = '>= 2.5'
 
   s.add_dependency 'aws-sigv4', '>= 1'
-  s.add_dependency 'opensearch-ruby', '>= 1.0.1', '< 4.0'
+  s.add_dependency 'opensearch-ruby', '>= 4.0.0.pre.beta.3'
```

Comment:
Since this is just a callback, not a wrapper for opensearch-ruby anymore. I think we can remove the dependency on this gem. It will reduce complexity for building and installing this gem.